File: 1168-duplicate-zeros/duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function (arr) {
    let ans = [];
    for(let i = 0; i < arr.length; i++){
        ans.push(arr[i]);
        if(arr[i] === 0){
            ans.push(0);
        }
    }
    for(let i = 0; i < arr.length; i++){
        arr[i] = ans[i];
    }
};

File: 1168-duplicate-zeros/duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        for(int i = 0; i < arr.length; i++){
            if(arr[i] == 0){
                shiftIt(arr,i);
                i++;
            } 
        }
    }
    public static void shiftIt(int arr[], int idx){
        for(int i = arr.length-1; i > idx; i--){
            arr[i] = arr[i-1];
        }
    }
}